config/software/libffi.rb

```diff
@@ -15,18 +15,23 @@
 #
 
 name "libffi"
-default_version "3.3"
+default_version "3.4.2"
 
 license "MIT"
 license_file "LICENSE"
 skip_transitive_dependency_licensing true
 
-# version_list: url=ftp://sourceware.org/pub/libffi/ filter=*.tar.gz
+# version_list: url=https://github.com/libffi/libffi/releases filter=*.tar.gz
 
+version("3.4.2") { source sha256: "540fb721619a6aba3bdeef7d940d8e9e0e6d2c193595bc243241b77ff9e93620" }
+version("3.3")   { source sha256: "72fba7922703ddfa7a028d513ac15a85c8d54c8d67f55fa5a4802885dc652056" }
```

Comment:
It looks like all the old releases we care about are also on GitHub so we can just update the SHAs for those releases and remove the if/else logic below that decides where to source